### PR TITLE
Mct4065

### LINKED ISSUE #4065: Clear data for objects clears data for all objects
#### Summary
`Clear data for objects` clears data for all objects

#### Expected vs Current Behavior
Should clear a data only for given object

#### Impact Check List
- [x] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug?

#### Steps to Reproduce
1. create / use this SWG on banner/dev -> [SWG](https://banner.ndc.nasa.gov/dev/#/browse/mine/cb56f6bf-c900-43b7-b923-2e3b64b412db/6fbeb716-440a-45c5-94fe-25e124e2dd82?tc.mode=local&tc.startDelta=900000&tc.endDelta=15000&tc.timeSystem=utc&view=plot-simple)
2. create empty overlay plot or use this -> [overlay plot](https://banner.ndc.nasa.gov/dev/#/browse/mine/cb56f6bf-c900-43b7-b923-2e3b64b412db/126c9be6-02ce-448e-ac66-f7ff6a67f080?tc.mode=local&tc.startDelta=900000&tc.endDelta=15000&tc.timeSystem=utc&view=plot-overlay) 
3. select SWG to show some data [SWG](https://banner.ndc.nasa.gov/dev/#/browse/mine/cb56f6bf-c900-43b7-b923-2e3b64b412db/6fbeb716-440a-45c5-94fe-25e124e2dd82?tc.mode=local&tc.startDelta=900000&tc.endDelta=15000&tc.timeSystem=utc&view=plot-simple)
3. context click on empty overlay plot and select  `Clear data for objects`
              Screen Shot 2021-07-28 at 12 48 04 PM
4. observe data shown for SWG is now cleared.

#### Environment
* Open MCT Version: 
                Version: 1.7.4
                Build Date: Thu Jul 22 2021 11:35:01 GMT-0700 (Pacific Daylight Time)
                Revision: 42af3ad2fccff666cc1749c89f49a0a9849cbb90
                Branch: flexible-inspector-properties
* Deployment Type: dev, viper
* OS: MAC OS 
* Browser: Brave

#### Additional Information
NA

### PR DESCRIPTION
**Pull request by scottbell**
fixes #4065 (comment)

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [X] Changes address original issue?
* [X] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue?

### CODE REVIEW

File: src/plugins/clearData/test/ClearDataActionSpec.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    it('clears data on applicable objects', () => {
+        let action = new ClearDataAction(openmct, ['apple']);
+
+        const actionApplies = action.appliesTo(mockObjectPath);
+
+        expect(actionApplies).toBe(true);
+    });
+
+    it('does not clear data on inapplicable objects', () => {
+        let action = new ClearDataAction(openmct, ['pineapple']);
```
[the remark this answers is not in the document]
Comment (scottbell) [reply src/plugins/clearData/test/ClearDataActionSpec.js:100]: 🍍

File: src/plugins/plot/MctPlot.vue [hunk 2]
```diff
@@ -399,22 +399,21 @@ export default {
...
-                    }
-                }
-            } else {
-                // global clear, fire it anyway!
+            // If we don't have an object to clear (global), or the IDs are equal, just clear the data.
+            // If we have an object to clear, but the IDs don't match, we need to check the composition
+            // of the object we've been asked to clear to see if it contains the id we're looking for.
+            // This happens with stacked plots for example.
+            // If we find the ID, clear the plot.
+            if (!domainObjectToClear
```
Comment (nikhilmandlik) [inline src/plugins/plot/MctPlot.vue:414]: 👍

**Review by nikhilmandlik — APPROVED** [review]
LGTM

**Comment by nikhilmandlik** [comment]
### Reviewer Checklist
Check | Passed
-- | --
Changes appear to address issue? | Y
Appropriate unit tests included? | Y
Code style and in-line documentation are appropriate? | Y
Commit messages meet standards? | Y
Has associated issue been labelled unverified? (only applicable if this PR closes the issue) | Y

**Pull request merged**